dsp-shared/src/test/scala/dsp/valueobjects/SchemaSpec.scala

```diff
@@ -0,0 +1,483 @@
+      assertTrue(
+        Schema.GuiAttribute.make(guiAttributeUnknownString) == Validation.fail(
+          BadRequestException(SchemaErrorMessages.GuiAttributeUnknown)
+        )
+      )
+    },
+    test("pass a valid value with whitespace and successfully create value object") {
+      assertTrue(
+        Schema.GuiAttribute
+          .make(guiAttributeSizeStringWithWhitespace)
+          .toOption
+          .get
+          .k == guiAttributeSizeStringKey
+      ) &&
+      assertTrue(
+        Schema.GuiAttribute
+          .make(guiAttributeSizeStringWithWhitespace)
+          .toOption
+          .get
+          .v == guiAttributeSizeStringValue
+      )
+    },
+    test("pass a valid value and successfully create value object") {
+      assertTrue(Schema.GuiAttribute.make(guiAttributeSizeString).toOption.get.k == guiAttributeSizeStringKey) &&
+      assertTrue(Schema.GuiAttribute.make(guiAttributeSizeString).toOption.get.v == guiAttributeSizeStringValue) &&
+      assertTrue(Schema.GuiAttribute.make(guiAttributeSizeString).toOption.get.value == guiAttributeSizeString)
+    }
+  )
+
+  private val guiElementTest = suite("gui element")(
+    test("pass an empty value and return an error") {
+      assertTrue(
+        Schema.GuiElement.make("") == Validation.fail(BadRequestException(SchemaErrorMessages.GuiElementMissing))
+      )
+    },
+    test("pass an unknown value and return an error") {
+      assertTrue(
+        Schema.GuiElement.make(guiElementUnknownIri) == Validation.fail(
+          BadRequestException(SchemaErrorMessages.GuiElementUnknown)
+        )
+      )
+    },
+    test("pass a valid value and successfully create value object") {
+      assertTrue(Schema.GuiElement.make(guiElementListIri).toOption.get.value == guiElementListIri)
+    }
+  )
+
+  private val validateGuiObjectsPointingToListTest = suite("validateGuiObjectsPointingToList")(
+    test(
+      "pass gui element 'salsah-gui#List' with gui attribute 'hlist' and successfully create value object"
+    ) {
+      val guiElement    = guiElementList
+      val guiAttributes = scala.collection.immutable.List(guiAttributeHlist)
+      val result        = Schema.validateGuiObjectsPointingToList(guiElement, guiAttributes)
+      assertTrue(
+        result == Validation.succeed(guiAttributes)
+      )
+    },
+    test(
+      "pass a gui element that points to a list but has a misfitting gui attribute and return an error"
+    ) {
+      val guiElement    = guiElementList
+      val guiAttributes = scala.collection.immutable.List(guiAttributeSize)
+      assertTrue(
+        Schema.validateGuiObjectsPointingToList(guiElement, guiAttributes) == Validation.fail(
+          BadRequestException(
+            "salsah-gui:guiAttribute for salsah-gui:guiElement GuiElement(http://www.knora.org/ontology/salsah-gui#List) has to be a list reference of the form 'hlist=<LIST_IRI>', but found GuiAttribute(size,80)."
+          )
+        )
```

Comment:
I tend to like the pattern of first declaring a variable with the expected result and then asserting against that, rather than writing it only in the assertion. In most cases I find it more readable. But that's of course 100% a matter of taste.